jboss-eap-cd-openshift-launch/added/launch/messaging.sh

```diff
@@ -513,7 +513,7 @@ function inject_brokers() {
   fi
 
   defaultJms=""
-  if [ -n "$defaultJmsConnectionFactoryJndi" ]; then
+  if [ -n "$defaultJmsConnectionFactoryJndi" &&  "x${DISABLE_EMBEDDED_JMS_BROKER}" != "xtrue" ]; then
```

Comment:
I think this check needs to be more complex.

The simple one is if REMOTE_AMQ_BROKER=true then I think this attribute still needs to be set regardless of DISABLE_EMBEDDED_JMS_BROKER.

The more complex and more of a corner-case is that in configure_mq if [ -n ${destinations} ] then things are configured regardless of the DISABLE_EMBEDDED_JMS_BROKER setting, so this should be consistent. That's kind of a corner case now but I expected in CD 16 or 17 we would change the DISABLE_EMBEDDED_JMS_BROKER default to 'true', and then it would be less of a corner case.

Reply:
Current iteration of this looks ok. In the future we can look at simplifying that a bit; e.g. calling configure_mq_destinations, storing the output in a var and passing it as a param to both inject_brokers and configure_mq. Or something.